lib/safira_web/views/redeemable_view.ex

```diff
@@ -0,0 +1,37 @@
+defmodule SafiraWeb.RedeemableView do
+  use SafiraWeb, :view
+  alias SafiraWeb.RedeemableView
+  alias Safira.Avatar
+
+  def render("index.json", %{redeemables: redeemables}) do
+    %{data: render_many(redeemables, RedeemableView, "redeemables.json")}
+  end
+
+  def render("show.json", %{redeemable: redeemable}) do
+    %{data: render_one(redeemable, RedeemableView, "redeemable_show.json")}
+  end
+
+  def render("redeemables.json", %{redeemable: redeemable}) do
+    %{
+      id: redeemable.id,
+      name: redeemable.name,
+      description: redeemable.description,
+      image: Avatar.url({redeemable.img, redeemable}, :original),
+      price: redeemable.price,
+      redeem_limit: redeemable.redeem_limit,
+      max_per_user: redeemable.max_per_user
+    }
+  end
+
+  def render("redeemable_show.json", %{redeemable: redeemable}) do
+    %{
+      id: redeemable.id,
+      name: redeemable.name,
+      description: redeemable.description,
+      image: Avatar.url({redeemable.img, redeemable}, :original),
+      price: redeemable.price,
+      redeem_limit: redeemable.redeem_limit,
+      max_per_user: redeemable.max_per_user
+    }
+  end
```

Comment:
This 2 renders are exactly the same. Do we need both?

Reply:
Responding to your questions:
1. I have put it so that you can easily know how many items have already been redeemed, I could decrement the limit but by doing so the defined limit would be lost, is that expected?
2. In this PR comment, I said that the buy logic is not final, I started it just to have an headstart. Answering directly, I'm thinking of adding a var in the buy table, named (e.g. 'num', 'redeemed' or something else) that tracks that number. So no, it is not my intention to have an entry for each buy. 
3. Regarding render, I didn't know if the behaviour of rendering all items was the same as rendering just one, so I left it like that. If the behaviour is the same, I delete it, if not I change it.

Reply:
1. That limit is the stock, I think there is no problem in decrement it.
2. Ok, but don't forget that later.
3. That diference is in the functions [render_many](https://hexdocs.pm/phoenix/Phoenix.View.html#render_many/4) and [render_on](https://hexdocs.pm/phoenix/Phoenix.View.html#render_one/4). The repeated render functions that you have just define the templates to be returned. But test it to see what i'm saying.

Reply:
adding to this, normally we have two different render functions where one is called by the :index action and the other by the :show, the :show one is usually more rich in content since we're examining one entry in detail while the :index, since it shows a list of entries, usually is less detailed. 

In the specific case of a store, usually, when seeing the list of all products, only the name, image and price is displayed, while when examining a specific product will show many more details about it.

Reply:
1. If there's no problem, I delete that field, I can change the name of 'redeem_limit' to 'stock' if it is clearer to you.

3. I understand what you are saying, my doubt was "If for store(when all items appear - 'render_many') only the name, price and image (example) appear, and in the view of an item (render_one) it shows all the fields". So we always send all the fields and the frontend shows only what matters?

Reply:
If the renders show different information then we need both, if they show the same information one render is enough. Normally, as @jpcorreia99 said, there is a distinction, but if in this case there is no distinction then one render is enough.